io-engine/src/bdev/device.rs

```diff
@@ -533,57 +531,43 @@ impl BlockDeviceHandle for SpdkBlockDeviceHandle {
 
         Ok(0)
     }
-    /// Flush the io in buffer to disk, for the Local Block Device.
-    async fn flush_io(&self) -> Result<u64, CoreError> {
-        if !self.device.0.is_write_cache_enabled() {
-            debug!(
-                "No Write Cache, No need of Flush for bdev name: {}, uuid: {}",
-                self.device.device_name(),
-                self.device.uuid(),
-            );
-            return Ok(0);
-        }
+    // Flush the io in buffer to disk, for the Local Block Device.
+    fn flush_io(
+        &self,
+        cb: IoCompletionCallback,
+        cb_arg: IoCompletionCallbackArg,
+    ) -> Result<(), CoreError> {
+        let ctx = alloc_bdev_io_ctx(
+            IoType::Flush,
+            IoCtx {
+                device: self.device,
+                cb,
+                cb_arg,
+            },
+            0,
+            0,
+        )?;
+
         let (desc, chan) = self.handle.io_tuple();
         let bdev_size = self.device.size_in_bytes();
-        let (s, r) = oneshot::channel::<bool>();
-        let result = unsafe {
+        let rc = unsafe {
             spdk_bdev_flush(
```

Comment:
Do We need to check for bdev volatile cache before calling spdk_flush ? if there is no cache, can skip the flush operation ?

Reply:
Good point.
We can add such an optimisation here.